python/ray/data/datasource/parquet_datasource.py

```diff
@@ -117,60 +121,6 @@ def _deserialize_fragments(
     return [p.deserialize() for p in serialized_fragments]
 
 
-def _deserialize_fragments_with_retry(
-    serialized_fragments: List[_SerializedFragment],
-) -> List["pyarrow._dataset.ParquetFileFragment"]:
-    """
-    Deserialize the given serialized_fragments with retry upon errors.
-
-    This retry helps when the upstream datasource is not able to handle
-    overloaded read request or failed with some retriable failures.
-    For example when reading data from HA hdfs service, hdfs might
-    lose connection for some unknown reason expecially when
-    simutaneously running many hyper parameter tuning jobs
-    with ray.data parallelism setting at high value like the default 200
-    Such connection failure can be restored with some waiting and retry.
-    """
-    min_interval = 0
-    final_exception = None
-    for i in range(FILE_READING_RETRY):
-        try:
-            return _deserialize_fragments(serialized_fragments)
-        except Exception as e:
-            import random
-            import time
-
-            retry_timing = (
-                ""
-                if i == FILE_READING_RETRY - 1
-                else (f"Retry after {min_interval} sec. ")
-            )
-            log_only_show_in_1st_retry = (
-                ""
-                if i
-                else (
-                    f"If earlier read attempt threw certain Exception"
-                    f", it may or may not be an issue depends on these retries "
-                    f"succeed or not. serialized_fragments:{serialized_fragments}"
-                )
-            )
-            logger.exception(
-                f"{i + 1}th attempt to deserialize ParquetFileFragment failed. "
-                f"{retry_timing}"
-                f"{log_only_show_in_1st_retry}"
-            )
-            if not min_interval:
-                # to make retries of different process hit hdfs server
-                # at slightly different time
-                min_interval = 1 + random.random()
-            # exponential backoff at
-            # 1, 2, 4, 8, 16, 32, 64
-            time.sleep(min_interval)
-            min_interval = min_interval * 2
```

Comment:
the retry backoff here is different from `call_with_retry()`, but looks fine to use the `call_with_retry()` logic.

Reply:
Yeah. The logic is similar enough. Made sense to deduplicate.